.github/workflows/ci.yml

```diff
@@ -0,0 +1,21 @@
+---
+name: ci
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+  pull_request:
+
+concurrency:
+  group: ci-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}
+  cancel-in-progress: true
+
+jobs:
+  prechecks:
+    uses: ./.github/workflows/pre-commit.yml
+  all-prechecks:
+    needs: [prechecks]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Success
+        run: "true"
```

Comment:
not sure I understand why this needs to be here? Isnt this doing nothing?

Reply:
Can you give me a little more context? This PR gets sorting working with pre-commit.ci and adds the framework for the switch to gha? If so LGTM, if not then I need a little more context

Reply:
This PR runs both pre-commit.ci (without sorting) and the "old" way of regular GHA (with sorting).  The advantage to pre-commit.ci is the bot PRs that update the hook versions, but I'm eventually going to switch to Renovate, which does that and replaces Dependabot.

Reply:
> not sure I understand why this needs to be here? Isnt this doing nothing?

It technically does nothing and just marks the final status of the workflow.  See https://github.com/cclib/cclib/blob/110292d71d4a88b3e1c2ec37c5fc95d4abf5b4b7/.github/workflows/ci.yml#L24.